FISH 6615 fix payaramicro tck for payara micro

simonladen opened the pull request:
Updating the Microprofile TCK runner to use the payara-micro-remote profile 
Caveats:
- Microprofile Config has 2 tests failing --> FISH-7050
- Microprofile JWT-Auth has 2 failing tests --> FISH-7051. 
- Microprofile Fault Tolerance - it runs a subset of the tests, excluding 40 additional failing tests --> FISH-7060
Example of results in Jenkins with these changes: https://jenkins.payara.fish/job/MPTCK-Dev/48/

Pandrex247 reviewed (commented):
Some comments:

* There should probably also be another follow-up task to enable the MicroProfile-Rest-Client/tck-runners/apache-http-client for Payara Micro.
Along with fixing the two issues you've already created, this would be another thing required for us to get Payara Micro certified.
* Another follow-up improvement would be to skip the start and stop of the Payara Micro instance when `-DskipTests` is enabled 🙂

Pandrex247 commented on MicroProfile-Config/pom.xml:
```diff
@@ -193,5 +193,59 @@
+                    <plugin>
+                        <groupId>fish.payara.maven.plugins</groupId>
+                        <artifactId>payara-micro-maven-plugin</artifactId>
+                        <version>1.0.6</version>
```
It would be nice if this version was defined as a property, and for the version to be managed via a `pluginManagement` section in the top-level pom. This would allow us to adjust the version on the fly as a command line option, and for us to only have to update the version in one place when it comes to alter it

Pandrex247 commented on MicroProfile-Fault-Tolerance/pom.xml:
```diff
@@ -168,5 +168,60 @@
+                    <plugin>
+                        <groupId>fish.payara.maven.plugins</groupId>
+                        <artifactId>payara-micro-maven-plugin</artifactId>
+                        <version>1.0.6</version>
```
Same comment as for Config.

Pandrex247 commented on pom.xml:
```diff
@@ -258,9 +258,84 @@
+                    <plugin>
+                        <groupId>fish.payara.maven.plugins</groupId>
+                        <artifactId>payara-micro-maven-plugin</artifactId>
+                        <version>1.0.6</version>
```
See comment for Config

Pandrex247 reviewed (approved):
It would be nice if this didn't leave a load of instances running in the background.

Numerous other iterative improvements can also be made but I'll raise those separately since they're more problems with the connector IMO (the concept of _remotely_ testing Payara Micro has always sat oddly with me as its not frankly the intended usage of it)

simonladen commented:
> WIP review comments:
> 
> * There should probably also be another follow-up task to enable the MicroProfile-Rest-Client/tck-runners/apache-http-client for Payara Micro.
>   Along with fixing the two issues you've already created, this would be another thing required for us to get Payara Micro certified.
> * Another follow-up improvement would be to skip the start and stop of the Payara Micro instance when `-DskipTests` is enabled 🙂
> 
> All non-blocker, just thoughts.

Issues FISH-7061 and FISH-7062 added in the backlog for these 2 tasks. 
I've also added FISH-7060 to look into running the full tck suite for Fault Tolerance (instead of the current "tck-suite3.0-stable.xml" which excludes about 40 tests)

I'll make the change for the version of payara-micro-maven-plugin

simonladen commented:
I believe that I have addressed your comments for the payara plugin version, and I have added steps to stop payara-micro in the pom files restarting it with additional parameters. Running the tests locally, it seems that I have no longer a hanging jdk process. Hopefully that addresses the issue.
